can/bus.py

```diff
@@ -141,4 +143,13 @@ def shutdown(self):
         """
         self.flush_tx_buffer()
 
-    __metaclass__ = abc.ABCMeta
+    @property
+    @abc.abstractmethod
+    def state(self):
+        return 'Should never read this'
```

Comment:
Instead of the unreachable code add a docstring - it will serve a dual purpose of making the method valid and documenting what the state will be for users/implementors.